src/xrServerEntities/smart_cast.h

```diff
@@ -15,11 +15,12 @@
 
 #define  TL_FAST_COMPILATION
 #undef   STATIC_CHECK
-#include <loki/typelist.h>
```

Comment:
TL_FAST_COMPILATION and STATIC_CHECK must been moved with typelist include. Otherwise compilation fails because STATIC_CHECK is not defined.